fortran/mpif-h: fix bindings for collective operations

ggouaillardet opened the pull request.

ggouaillardet committed:
checkpoint

ibm-ompi commented:
The IBM CI (GNU Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/ibm-ompi/88b480fb98dfd2bdac939f023e1f8fe6

ggouaillardet committed:
fortran/use-mpi-f08: generates PMPI bindings from the MPI bindings

 - via the OMPI_BUILD_MPI_PROFILING macro, avoid the need to have two distinct
   source files for MPI_* and PMPI_* bindings.
 - move the PMPI bindings into ompi/mpi/fortran/use-mpi-f08/profile.
 - remove an useless dependency to mpi-f08.lo

Signed-off-by: Gilles Gouaillardet <gilles@rist.or.jp>

ibm-ompi commented:
The IBM CI (GNU/Scale) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/81c7ba21315613c00140b5ea80142c8b

lanl-ompi commented:
Can one of the admins verify this patch?

hppritcha commented:
closing.  can be reopened if someone wishes to pursue this.
